src/test/java/com/walmartlabs/x12/util/split/X12TransactionSplitterTest.java

```diff
@@ -0,0 +1,426 @@
+        
+        // verify 2nd document
+        String expectedDocTwo = new StringBuilder()
+            .append("ISA*01*0000000000*01*0000000000*ZZ*ABCDEFGHIJKLMNO*ZZ*123456789012345*101127*1719*U*00400*000000049*0*P*>")
+            .append("\r\n")
+            .append("GS*SH*4405197800*999999999*20111206*1045*00*X*004060")
+            .append("\r\n")
+            .append("ST*BBB*0002")
+            .append("\r\n")
+            .append("TEST*2")
+            .append("\r\n")
+            .append("SE*1*0002")
+            .append("\r\n")
+            .append("GE*1*00")
+            .append("\r\n")
+            .append("IEA*2*000000049")
+            .toString();
+        String docTwo = ediDocuments.get(1);
+        assertEquals(expectedDocTwo, docTwo);
+        
+        // verify 3rd document
+        String expectedDocThree = new StringBuilder()
+            .append("ISA*01*0000000000*01*0000000000*ZZ*ABCDEFGHIJKLMNO*ZZ*123456789012345*101127*1719*U*00400*000000049*0*P*>")
+            .append("\r\n")
+            .append("GS*SH*4405197800*999999999*20111206*1045*00*X*004060")
+            .append("\r\n")
+            .append("ST*YYZ*0099")
+            .append("\r\n")
+            .append("TEST*99")
+            .append("\r\n")
+            .append("SE*1*0099")
+            .append("\r\n")
+            .append("GE*1*00")
+            .append("\r\n")
+            .append("IEA*2*000000049")
+            .toString();
+        String docThree = ediDocuments.get(2);
+        assertEquals(expectedDocThree, docThree);
+        
+        // verify 4th document
+        String expectedDocFour = new StringBuilder()
+            .append("ISA*01*0000000000*01*0000000000*ZZ*ABCDEFGHIJKLMNO*ZZ*123456789012345*101127*1719*U*00400*000000049*0*P*>")
+            .append("\r\n")
+            .append("GS*SH*4405197800*999999999*20111206*1045*99*X*004060")
+            .append("\r\n")
+            .append("ST*AAA*0003")
+            .append("\r\n")
+            .append("TEST*3")
+            .append("\r\n")
+            .append("SE*1*0003")
+            .append("\r\n")
+            .append("GE*1*99")
+            .append("\r\n")
+            .append("IEA*2*000000049")
+            .toString();
+        String docFour = ediDocuments.get(3);
+        assertEquals(expectedDocFour, docFour);
+    }
+    
+    @Test(expected = X12ParserException.class)
+    public void test_split_sourceData_one_group_one_document_rules_fail() throws IOException {
+        byte[] x12Bytes = Files.readAllBytes(Paths.get("src/test/resources/x12.base.one.txt"));
+        String sourceData = new String(x12Bytes);
+        
+        X12Rule mockRule = Mockito.mock(X12Rule.class);
+        Mockito.doThrow(X12ParserException.class)
+            .when(mockRule)
+            .verify(Mockito.anyList());
+        
+        splitter.registerX12Rule(mockRule);
+        
+        splitter.split(sourceData);
+    }
+    
+    @Test(expected = X12ParserException.class)
```

Comment:
This test should not raise any exception I think

Reply:
will look at that once PR #86  is merged

Reply:
next commit will address this...

Reply:
done